crates/task-impls/src/consensus/proposal_helpers.rs

```diff
@@ -566,6 +567,12 @@ pub async fn publish_proposal_if_able<TYPES: NodeType>(
     }
 }
 
+#[cfg(feature = "dependency-tasks")]
+type TempraryProposalCombinedType<TYPES, I, A> = QuorumProposalRecvTaskState<TYPES, I>;
+
+#[cfg(not(feature = "dependency-tasks"))]
+type TempraryProposalCombinedType<TYPES, I, A> = ConsensusTaskState<TYPES, I, A>;
```

Comment:
Curious why we need these types. Looks like they are the same as the existing `*State` structs. Will this change later?